src/Microsoft.DotNet.ProjectModel/OutputPathCalculator.cs

```diff
@@ -12,142 +11,103 @@ namespace Microsoft.DotNet.ProjectModel
     public class OutputPathCalculator
     {
         private const string ObjDirectoryName = "obj";
+        private const string BinDirectoryName = "bin";
 
         private readonly Project _project;
         private readonly NuGetFramework _framework;
 
-        private readonly string _runtimeIdentifier;
-
-        /// <summary>
-        /// Unaltered output path. Either what is passed in in the constructor, or the project directory.
-        /// </summary>
-        private string BaseOutputPath { get; }
-
-        public string BaseCompilationOutputPath { get; }
+        private readonly string _runtimePath;
+        private readonly RuntimeOutputFiles _runtimeFiles;
 
         public OutputPathCalculator(
             Project project,
             NuGetFramework framework,
             string runtimeIdentifier,
-            string baseOutputPath)
+            string configuration,
```

Comment:
This is not good. There is not way that someone trying to use this class can figure out what to pass to all these paths without finding a current usage for the class and reverse engineering that. We need to A, come up with a slightly different design for this, or B, come up with better names for these properties/parameters.